go/apps/dialogue/tests/test_vumi_app.js

```diff
@@ -318,5 +320,68 @@ describe("app", function() {
                     });
             });
         });
+
+        describe("when the user enters a send state which is a httpjson type", function() {
+            it("should successfully fetch the data from the url if the method is GET", function() {
+                return tester
+                    .setup(function(api) {
+                        api.contacts.add({
+                           msisdn: '+27123'
+                        });
+                        api.http.fixtures.add({
+                            request: {
+                                method: "GET", 
+                                url: "www.foo.bar"
+                            }
+                        });
+                    })
+                    .setup.user.addr('+27123')
+                    .setup.user.state('choice-1')
+                    .input('4')
+                    .check(function(api) {
+                        assert.equal(api.http.fixtures.fixtures[0].responses[0].code, 200);
+                    })
```

Comment:
I think we should also check that the end state was reached (i.e. that the person was successfully routed through to the next state).